File: daily-challenge/2025-09-22/solution.py
"""
Problem: 3005 - Count Elements With Maximum Frequency  
Description: Given an array of positive integers, return the total number of elements whose frequency equals the maximum frequency in the array.  
Date: 2025-09-22
"""

from collections import Counter

class Solution:
    def maxFrequencyElements(self, nums):
        freq = Counter(nums)
        max_freq = max(freq.values())
        return sum(count for count in freq.values() if count == max_freq)

File: daily-challenge/2025-09-22/solution.cpp
"""
Problem: 3005 - Count Elements With Maximum Frequency  
Description: Given an array of positive integers, return the total number of elements whose frequency equals the maximum frequency in the array.  
Date: 2025-09-22
"""

#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    int maxFrequencyElements(vector<int>& nums) {
        unordered_map<int, int> freq;
        int maxFreq = 0, total = 0;

        for (int num : nums) {
            freq[num]++;
            maxFreq = max(maxFreq, freq[num]);
        }

        for (auto& [_, count] : freq) {
            if (count == maxFreq) total += count;
        }

        return total;
    }
};
